### PR TITLE
{numlib}[intel/2021b] SuiteSparse v5.8.1, PETSc v3.18.4, SLEPc v3.18.2 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name = 'SuiteSparse'
+version = '5.8.1'
```
Comment (branfosj) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb:2]: We already have SuiteSparse 5.10.1 (https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-foss-2021b-METIS-5.1.0.eb) in the `foss/2021b` toolchain. Can we match that version here for the `intel/2021b` toolchain?

**Review by branfosj — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb:2]: For
```
icc: error #10106: Fatal error in /kyukon/home/apps/RHEL8/cascadelake-ib/software/intel-compilers/2021.4.0/compiler/2021.4.0/linux/bin/intel64/../../bin/intel64/mcpcom, terminated by kill signal
```
my first guess is out of memory.

I tried a test build and I also failed in the same way, but I'd got to 7% of that section done. I am now trying with more memory and fewer cores.

Comment (branfosj) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb:2]: So yes, with more memory the installation completes. I saw it peak at 125GB when building!

Comment (ItIsI-Orient) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb:2]: I see, should I use `parallel = 1` in the easyconfig just to ease things out?

Comment (branfosj) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb:2]: For SuiteSparse that will not help. The issue is the generation of an optimisation report. See https://github.com/easybuilders/easybuild-easyconfigs/pull/17285#issuecomment-1474125285

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+dependencies = [
+    ('METIS', local_metis_ver),
+    ('MPFR', '4.1.0'),
+]
+
+parallel = 1
+# make sure that bin/demo can find libsuitesparseconfig.so.5 during build
+prebuildopts = "export LD_LIBRARY_PATH=%(builddir)s/SuiteSparse-%(version)s/lib:$LD_LIBRARY_PATH && "
+# remove broken symlink
+# prebuildopts += "rm GraphBLAS/CUDA/test/graphblascuda_test && "
```
Comment (branfosj) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb:30]:
```suggestion
# disable optimisation report to reduce memory use during compile
prebuildopts += 'sed -i "s/-qopt-report=5/-qopt-report=0/" GraphBLAS/CMakeLists.txt && '
```

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.8.1-intel-2021b-METIS-5.1.0.eb]: c1187ae

**Comment by branfosj** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    ('CMake', '3.21.1'),
+    ('M4', '1.4.19'),
+]
+
+dependencies = [
+    ('METIS', local_metis_ver),
+    ('MPFR', '4.1.0'),
+]
+
+parallel = 1
```
Comment (branfosj) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb:26]:
```suggestion
```

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 6 out of 6 (3 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/e1e5cd834e70d36b5e19f616a810ec24 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @ItIsI-Orient!

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+dependencies = [
+    ('METIS', local_metis_ver),
+    ('MPFR', '4.1.0'),
+]
+
+# make sure that bin/demo can find libsuitesparseconfig.so.5 during build
+prebuildopts = "export LD_LIBRARY_PATH=%(builddir)s/SuiteSparse-%(version)s/lib:$LD_LIBRARY_PATH && "
+# disable optimisation report to reduce memory use during compile
+prebuildopts += 'sed -i "s/-qopt-report=5/-qopt-report=0/" GraphBLAS/CMakeLists.txt && '
```
[the remark this answers is not in the document]
Comment (branfosj) [reply easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021b-METIS-5.1.0.eb:29]: I've asked the question (https://github.com/DrTimothyAldenDavis/GraphBLAS/issues/211) to better understand if there are implications. Ideally it'd be changed upstream, meaning we'd not need this.

**Pull request merged**